src/uucore/src/lib/features/parser/num_parser.rs

```diff
@@ -500,10 +504,15 @@ mod tests {
     fn test_decimal_u64() {
         assert_eq!(Ok(123), u64::extended_parse("123"));
         assert_eq!(Ok(u64::MAX), u64::extended_parse(&format!("{}", u64::MAX)));
-        // TODO: We should wrap around here #7488
+        assert_eq!(Ok(0), u64::extended_parse("-0"));
+        assert_eq!(Ok(u64::MAX), u64::extended_parse("-1"));
+        assert_eq!(
+            Ok(u64::MAX / 2 + 1),
+            u64::extended_parse("-9223372036854775808")
+        );
         assert!(matches!(
-            u64::extended_parse("-123"),
-            Err(ExtendedParserError::Overflow(0))
+            u64::extended_parse("-9223372036854775809"),
+            Err(ExtendedParserError::Overflow(_))
```

Comment:
That doesn't match GNU coreutils behaviour:
```
env printf "%u" -9223372036854775809
9223372036854775807
```

Looks like numbers down to 2*i64::MIN still overflow.
```
env printf "%u" -17323372036854876000
1123372036854675616
```

Reply:
`-18446744073709551615` is `-u64::MAX` (`0xffffffffffffffff`).

I don't really know the logic (and we're trying not to look at glibc code), but I'd assume the number is just parsed into an `u64`, and then the signed is handled (taking the 2 complement would lead to a value of `1`).

(the explanation here doesn't seem to match what is happening: https://www.gnu.org/software/libc/manual/html_node/Parsing-of-Integers.html#index-strtoul)